### PR TITLE
MC-30127: Product Price is mismatch in invoice and invoice PDF magento for Bundle Product

### LINKED ISSUE #12856: Product Price is mismatch in invoice and invoice PDF magento for Bundle Product  2.2.1
I am getting different price in invoice and invoice PDf for the bundle product. Actually, The price in the invoice pdf are wrong. I am getting right price in order, order email, invoice, invoice email or everywhere in the application, but invoice PDf price is wrong only Bundle product. 

### Preconditions
1. Install Magento 2.2.1
1. Create Bundle product in application.
1. In Magento Admin go to Stores - Tax Rules and Add New Tax Rule
34373744-2e6633ec-eb03-11e7-9fbe-2aa2ffa9a13c
1. Go to Stores - Configuration - Sales - Tax and enable the following settings 
34373758-3a995126-eb03-11e7-857f-3b38d4fbf36c

### Steps to reproduce
1. Create the order with bundle product.
1. Create the invoice, then download the invoice PDF.

### Expected result

1. Price in the invoice PDF should be same as invoice page or order page.
Invoice Screen shot:

screenshot from 2017-12-22 14-38-25

### Actual result
1. Prices in the invoice PDF and invoice page are different. Getting wrong price in invoice PDF for bundle product. 
Invoice PDF Screen shot:
invoice-pdf

### PR DESCRIPTION
**Pull request by engcom-Golf**
### Fixed Issues (if relevant)

1. Fixes magento/magento2#12856

### Manual testing scenarios (*)

1. Create tax rule for CA
1. Create bundle, simple, downloadable product
2. Create order with these products for CA shipping address
3. Invoice order from admin
4. Configure Taxes to show in invoices(test different scenarios)
     Stores->Configuration->Sales->Tax->Orders, Invoices, Credit Memos Display Settings 
     Display Taxes & Display Subtotal
5. Print Invoice
#### Expected result
Prices for all products in pdf file should be the same as in admin
All text should be drawn correctly (font, align, padding)

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds are green)

### CODE REVIEW

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by engcom-Golf** [comment]
@magento run all tests

File: app/code/Magento/Bundle/Model/Sales/Order/Pdf/Items/Invoice.php [hunk 1]
```diff
@@ -15,10 +16,12 @@
 use Magento\Framework\Registry;
 use Magento\Framework\Serialize\Serializer\Json;
 use Magento\Framework\Stdlib\StringUtils;
+use Magento\Sales\Model\Order\Pdf\Items\Invoice\PreparePriceLines;
 use Magento\Tax\Helper\Data;
 
 /**
  * Order invoice pdf default items renderer
+ * @SuppressWarnings(PHPMD.CouplingBetweenObjects)
```
Comment (engcom-Kilo) [inline app/code/Magento/Bundle/Model/Sales/Order/Pdf/Items/Invoice.php]: Please try to find different solution without adding new dependency.

**Review by engcom-Kilo — CHANGES_REQUESTED** [review]

**Comment by engcom-Golf** [comment]
@magento run all tests

**Comment by engcom-Golf** [comment]
@magento run all tests

File: app/code/Magento/Bundle/Model/Sales/Order/Pdf/Items/Invoice.php [hunk 2]
```diff
@@ -7,6 +7,7 @@
 
 namespace Magento\Bundle\Model\Sales\Order\Pdf\Items;
 
+use Magento\Framework\App\ObjectManager;
```
Comment (engcom-Kilo) [inline app/code/Magento/Bundle/Model/Sales/Order/Pdf/Items/Invoice.php]: Redundant dependency. Please remove.

File: app/code/Magento/Bundle/Model/Sales/Order/Pdf/Items/Invoice.php [hunk 3]
```diff
@@ -124,32 +122,64 @@ public function draw()
...
-                $line[] = ['text' => $tax, 'feed' => 495, 'font' => 'bold', 'align' => 'right'];
-
-                $row_total = $order->formatPriceTxt($childItem->getRowTotal());
-                $line[] = ['text' => $row_total, 'feed' => 565, 'font' => 'bold', 'align' => 'right'];
+                $lines[$index][] = ['text' => $tax, 'feed' => 495, 'font' => 'bold', 'align' => 'right'];
+
+                $this->_item = $childItem;
+                $feedPrice = 380;
+                $feedSubtotal = $feedPrice + 185;
+                foreach ($this->getItemPricesForDisplay() as $priceData) {
```
Comment (engcom-Kilo) [inline app/code/Magento/Bundle/Model/Sales/Order/Pdf/Items/Invoice.php]: Could you please try to refactor this piece? This method already has ExcessiveMethodLength suppress.

**Review by engcom-Kilo — CHANGES_REQUESTED** [review]

**Comment by engcom-Golf** [comment]
@magento run all tests

File: app/code/Magento/Bundle/Model/Sales/Order/Pdf/Items/Invoice.php (hunk 3 above)
Comment (engcom-Golf) [reply app/code/Magento/Bundle/Model/Sales/Order/Pdf/Items/Invoice.php]: refactored draw method

**Review by engcom-Kilo — APPROVED** [review]

**Review by engcom-Bravo — CHANGES_REQUESTED** [review]
:x: QA Failed

The Printed Invoice doesn't contain one of the products.

### Manual testing scenario:
1. Create a Bundle product with two options.
- One _dropdown_ option
- second _checkbox_ option containing two products
2. Place an order selecting 1 product from the first option, two products from the second option
options_bundle
3. Submit and Print the invoice.
4. Observe the invoice

**Actual Result**
The first product from the second option is not displayed in the printed invoice
option_failed

**Comment by engcom-Golf** [comment]
@magento run all tests

**Review by engcom-Kilo — APPROVED** [review]

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by engcom-Bravo** [comment]
### Manual testing scenario
1. From Admin Panel, Go to _Stores - Tax Rulles_, click **Add New Tax Rule**
2. **Add New Tax Rate**, for example, for:
- US Customers
- All regions (*)
- 16%
**Save Rule**
3. Go to _Stores - Configuration - Sales - Tax_. Expand **Orders, Invoices, Credit Memos Display Settings**, and set **Including Tax** for the following options:
- _Display Prices_
- _Display Subtotal_
- _Display Shipping Amount_
include
4. Go to _Catalog - Products_ and add a new **Bundle** product:
- _Drop-down_ option
- _Radio button_ option
- _Checkbox_ option
- _Multiple select_ option
5. **Place** a new order with a US Customer, configuring several options from the Bungle product. For example
configure bundle
6. Go to **Invoice** and **Submit Invoice**
7. Go to _Invoices_ tab and **View** the previously created Invoice
8. **Print** the Invoice
9. Compare the Invoice from Admin and the Printed document

**BEFORE** applying changes provided in the PR, the price in the printed invoice did not contain TAX. Thus, the _Proce_ was different 
price_print

**AFTER** switching to the PR, the price in the printed invoice is identic with the price in the Invoice from Admin Panel
after_identic

**Comment by zakdma** [comment]
@magento import code to https://github.com/magento-tsg/magento2ce

**Comment by magento-engcom-team** [comment]
@zakdma, an error occurred during the Pull Request import.

**Comment by magento-engcom-team** [comment]
@zakdma the branch with code successfully imported into`magento-tsg/magento2ce` repository. Branch name: `imported-magento-magento2-31455`.

**Pull request merged**